### PR TITLE
Slice plugin: self heal when asset etag/last-modified block mismatches

### PR DESCRIPTION
**Pull request by traeak**
This tries to make the slice plugin smarter when origins do bad things to assets without changing their path including:
* etag change
* last modified change
* content length change
* change no nocache
* asset removed

Process:

1. Reference slice fetched (either slice0 or first slice in request).
2. Second slice fetched -- mismatch detected
3. Refetch second slice (cache_range_requests X-Crr-Ims header)
    a. Match reference, continue to next slice
    a. Mismatch reference, abort client, refetch reference slice (X-Crr-Ims header)

### CODE REVIEW

File: plugins/experimental/slice/client.cc [hunk 1]
```diff
@@ -119,15 +126,30 @@ handle_client_req(TSCont contp, TSEvent event, Data *const data)
 void
 handle_client_resp(TSCont contp, TSEvent event, Data *const data)
 {
-  DEBUG_LOG("%p handle_client_resp %s", data, TSHttpEventNameLookup(event));
+  // DEBUG_LOG("%p handle_client_resp %s", data, TSHttpEventNameLookup(event));
```
Comment (randall) [inline plugins/experimental/slice/client.cc:129]: can this be removed if it's not needed?

File: doc/admin-guide/plugins/cache_range_requests.en.rst [hunk 2]
```diff
@@ -125,13 +125,13 @@ special header:
 
 .. code::
 
-    X-CRR-IMS: Tue, 19 Nov 2019 13:26:45 GMT
+    X-Crr-Ims: Tue, 19 Nov 2019 13:26:45 GMT
 
 When this header is provided and a `cache hit fresh` is encoutered the
```
Comment (randall) [inline doc/admin-guide/plugins/cache_range_requests.en.rst:130]: encountered

File: doc/admin-guide/plugins/slice.en.rst [hunk 3]
```diff
@@ -194,30 +216,59 @@ adding in range request headers ("Range: bytes=").  A special
...
+
+Any extra bytes at the end of the last block are consumed by the the
+Slice plugin to allow cache_range_requests to finish the block fetch to
+ensure the block is cached.
+
+Self Healing
+------------
+
+The slice plugin uses the very first slice as a reference slice which
+uses content-length and last-modifed and/or etags to ensure assembled
```
Comment (randall) [inline doc/admin-guide/plugins/slice.en.rst:238]: modified

File: plugins/experimental/slice/client.cc (hunk 1 above)
Comment (traeak) [reply plugins/experimental/slice/client.cc]: When troubleshooting the TSVIO's these are nice to have available.

**Comment by zwoop** [comment]
Is this something you need for 9.0.x? Or 9.1.x ?

**Comment by traeak** [comment]
9.0.x would be nice.

**Comment by traeak** [comment]
Update removed some of the error logs and also trimmed more off the autests (traffic_manager -> traffic_server)

**Comment by randall** [comment]
[approve ci autest]

**Comment by zwoop** [comment]
Backported to 9.0.1 in #7435

**Pull request merged**